Tests/SwiftParserTest/ExpressionTests.swift

```diff
@@ -897,3 +897,154 @@ final class MemberExprTests: XCTestCase {
     }
   }
 }
+
+final class StatementExpressionTests: XCTestCase {
+  func testIfExprInCoercion() {
```

Comment:
Do you think it would make sense to add a few `substructure` assertions here? It seems like we could easily get the structure wrong here in some cases (although I didn’t look too closely). I’ll leave it up to you to decide where substructure assertions might make sense.